test/NuGet.Clients.Tests/NuGet.PackageManagement.VisualStudio.Test/Feeds/InstalledAndTransitivePackageFeedTests.cs

```diff
@@ -236,5 +227,114 @@ public async Task SearchAsync_WithInstalledAndTransitivePackages_AlwaysInstalled
             lastItems.Should().BeInAscendingOrder(idComparer);
             lastItems.Should().AllBeOfType<TransitivePackageSearchMetadata>();
         }
+
+        [Fact]
+        public async Task ContinueSearchAsync_WhenMultitargetingAndEmptySearch_ReturnsOnePackagePerPackageIdAsync()
+        {
+            // Arrange
+            IEnumerable<PackageCollectionItem> installedPackages = new List<PackageCollectionItem>()
+            {
+                new PackageCollectionItem("packageA", NuGetVersion.Parse("1.0.0"), new IPackageReferenceContextInfo[] {}),
+                new PackageCollectionItem("packageA", NuGetVersion.Parse("2.0.0"), new IPackageReferenceContextInfo[] {}),
+                new PackageCollectionItem("packageA", NuGetVersion.Parse("3.0.0"), new IPackageReferenceContextInfo[] {}),
+            };
+            IEnumerable<PackageCollectionItem> transitivePackages = new List<PackageCollectionItem>()
+            {
+                new PackageCollectionItem("transitivePackageA", NuGetVersion.Parse("0.0.1"), new List<IPackageReferenceContextInfo>()
+                {
+                    new TransitivePackageReferenceContextInfo(new PackageIdentity("transitivePackageA", NuGetVersion.Parse("0.0.1")), NuGetFramework.Parse("net6.0"))
+                    {
+                        TransitiveOrigins = new []
+                        {
+                            new PackageReferenceContextInfo(new PackageIdentity("packageA", new NuGetVersion("3.0.0")), NuGetFramework.Parse("net6.0"))
+                        }
+                    },
+                }),
+                new PackageCollectionItem("transitivePackageA", NuGetVersion.Parse("0.0.2"), new List<IPackageReferenceContextInfo>()
+                {
+                    new TransitivePackageReferenceContextInfo(new PackageIdentity("transitivePackageA", NuGetVersion.Parse("0.0.2")), NuGetFramework.Parse("net6.0"))
+                    {
+                        TransitiveOrigins = new []
+                        {
+                            new PackageReferenceContextInfo(new PackageIdentity("packageA", new NuGetVersion("2.0.0")), NuGetFramework.Parse("net6.0"))
+                        }
+                    },
+                }),
+                new PackageCollectionItem("transitivePackageA", NuGetVersion.Parse("0.0.3"), new List<IPackageReferenceContextInfo>()
+                {
+                    new TransitivePackageReferenceContextInfo(new PackageIdentity("transitivePackageA", NuGetVersion.Parse("0.0.3")), NuGetFramework.Parse("net6.0"))
+                    {
+                        TransitiveOrigins = new []
+                        {
+                            new PackageReferenceContextInfo(new PackageIdentity("packageA", new NuGetVersion("1.0.0")), NuGetFramework.Parse("net6.0"))
```

Comment:
It looks all 3 packages are targeting tfm `net6.0`, should we assert different tfms for transitive origins?

Reply:
Fixed test case. Thanks!